application-analytics-default/src/main/java/com/xwiki/analytics/internal/MostViewedJsonNormaliser.java

```diff
@@ -0,0 +1,200 @@
+    public JsonNode normaliseData(String jsonString) throws JsonProcessingException
+    {
+        // I need to convert the string returned by matomo in a JSON to easily handle the processing of the nodes.
+        JsonNode jsonRoot = OBJECT_MAPPER.readTree(jsonString);
+        // Matomo may return several variants of JSON formats. In one scenario, when the period is set to
+        // day/week/month/year, it returns a JSON object with keys representing dates. The corresponding value for
+        // each key is an array of JSON objects, each of which represents a page. However, if the user sets the
+        // period parameter to "range" Matomo returns an array of JSON objects, with each JSON object representing
+        // a page. Due to these variations, I need to process the result from Matomo to create a normalized format.
+        // This normalized format is an array of JSON objects and each JSON object in this array will have a new
+        // field called 'date'. This 'date' field will be set to 'N/A' when Matomo returns an array instead of an
+        // object. In the 'processArrayNode' and 'processObjectNode' methods, I also modify the 'label' field to
+        // change it from the raw URL format to the page name.
+        if (jsonRoot.isArray()) {
+            processArrayNode(jsonRoot);
+        } else {
+            jsonRoot = processObjectNode(jsonRoot);
+        }
+        return jsonRoot;
+    }
+
+    /**
+     * Handle the scenario where Matomo returns an array of JSON objects. This function processes each entry to append
+     * an empty date to it.
+     *
+     * @param jsonNode an array of jsons
+     */
+    private void processArrayNode(JsonNode jsonNode)
+    {
+        for (JsonNode objNode : jsonNode) {
+            if (objNode.isObject()) {
+                ((ObjectNode) objNode).put(DATE, "");
+                if (objNode.has(URL)) {
+                    this.handleURLNode((ObjectNode) objNode);
+                }
+            }
+        }
+    }
+
+    /**
+     * Handle the scenario where Matomo returns an object with dates as keys and arrays of JSON objects as values. This
+     * function extracts the date from the key and adds it to each page. Ultimately, it returns an array of JSON objects
+     * instead of a single JSON object.
+     *
+     * @param jsonNode json object
+     * @return array of jsons
+     */
+    private ArrayNode processObjectNode(JsonNode jsonNode)
+    {
+        ArrayNode arrayNode = OBJECT_MAPPER.createArrayNode();
+        Iterator<String> fieldNames = jsonNode.fieldNames();
+        while (fieldNames.hasNext()) {
+            String date = fieldNames.next();
+            JsonNode childNode = jsonNode.get(date);
+            for (JsonNode objNode : childNode) {
+                if (objNode.isObject()) {
+                    ((ObjectNode) objNode).put(DATE, date);
+                    if (objNode.has(URL)) {
+                        this.handleURLNode((ObjectNode) objNode);
```

Comment:
Just a note for the future, in case this method will be used for other normalizers, to be sure that there aren't other nodes that contain a an `url`